### ISSUE TITLE
[BUG] Space station not docking modules, then crashes when I return.

### ISSUE THREAD

**Issue by MarkOfTheBeast** [zmaster587/AdvancedRocketry#1658]
[Please fill out the form below and delete the sections in square brackets after reading them]

## Version of Advanced Rocketry
>[1.12.2-1.6.0-162 universal]

>Why is this important?
>Knowing the version of Advanced Rocketry helps us narrow down an offending piece of code faster because we can easily determine what code is added and when.]

## Have you verified this is an issue in the latest unstable build
- [x] Y

>Why is this important?
>If you're not testing with the latest build, you may be encountering a bug that's already been fixed.
>Reporting bugs that have already been fixed wastes everyone's time.  (please look in open issues too, somebody else could have already reported it!)

## Version of LibVulpes
>1.12.2-0.4.1-1-universal

>Why is this important?
>Advanced rocketry depends on a library mod (LibVulpes), some errors can really be in that mod.  So knowing the verison helps us not only start looking for fixes there without you needing to anwer more questions, it also helps us know if you have a version mismatch.

## Version of Minecraft
>1.12.2

>Why is this important?
>Different versions of minecraft have different codebases for Advanced Rocketry.  Knowing the version of minecraft (and thus AR) helps us pinpoint issues quicker!

## Does this occur without other mods installed
- [ ] Y

>If Y, what is the MINIMUM set of mods required.

>Why is this important?  
>Sometimes interactions between mods cause issues.  While it can be a problem with Advanced Rocketry, sometimes it's not.  When there are a lot of mods, it can take bugtesters HOURS to go through the list to determine the root cause.  When there are a lot of bug reports, that can add DAYS to the time it takes to fix a large set of issues.

## Crash report or log or visualVM (if applicable)
http://pastebin.com is a good place to put them
crash reports that are put in the issue itself are
hard to read

https://pastebin.com/kyk31j2z

> If it's an issue involving lag, please submit a visualVM sampler snapshot.  Mods like lag goggles often do not provide the kind of information needed to debug lag.
> For more information on how to create a snapshot, please see "How to Run CPU Sampling" [here](http://greyfocus.com/2016/05/visualvm-sampling/).  Then export the snapshot as an .nps.  You may have to zip it since github is weird, then please upload it here.

>Why is this important?
>Full crashlogs contain valuable information like where exactly in the code the error occurred, what other mods may be running and what version of forge you have.  This helps us replicate and diagnose the issue FAR faster.

## Description of the problem
When I try to dock a new module to my space station, it does not get attached, the space station container that I built it in is deleted from the rocket's satellite bay, and then the game crashed when I tried to land the rocket back in the overworld.

>Why is this important?
>If we cannot reproduce the problem, we probably cannot fix it at all.

**Comment by zmaster587** [comment]
How exactly are you docking them?  What names are you using in the docking ports?  What are you putting on the rocket carrying the module up?  Are you going up on the same rocket as the module?  Does it have a guidance computer?

Would you be able to get me a copy of the crash report generated with only AR and libvulpes installed?

**Comment by MarkOfTheBeast** [comment]
I am going up in the rocket when I send a module up. It has a guidance computer and a satellite bay in it. When this happened, the rocket also started landing on top of the center of the station again instead of the designated docking pad.

**Comment by MarkOfTheBeast** [comment]
When I run it in an isolated instance, the mod works but that same error appears in the log file.

**Comment by zmaster587** [comment]
Does the guidance computer have a chip in it?  If so what's on the chip?

Can you post the log file from the isolated instance?

**Comment by zmaster587** [comment]
I'm not able to replicate the issue with the docking port as of yet, however I was able to get the stack trace to print though the game doesn't crash.

I have tested building a rocket containing a guidance computer, a seat, and a satellite bay.  
The station container contains a build with a docking port with the target name "test".

The space station in orbit contained a docking port with the name "test".  The station container was loaded into the satellite bay on the rocket.  A chip programmed with the target station was loaded into the guidance computer.  The rocket was then fueled and launched with my player onboard.  After reaching orbit, it descended and landed on the station.  The module I built had been moved in place as expected.

I have however been able to replicate the stacktrace upon landing.  The game does not crash, but an error is printed.  I will fix that.
